flycheck-dmd-dub.el

```diff
@@ -161,6 +162,31 @@ other lines besides the json object."
   "Return the output from dub with package description."
   (shell-command-to-string "dub describe"))
 
+(defun fldd--get-timestamp (file)
+  "Return the timestamp of FILE.
+If FILE does not exist, return nil."
+  (when (file-exists-p file)
+    (nth 5 (file-attributes file))))
+
+(defun fldd--set-variables (import-paths string-import-paths)
+  "Set IMPORT-PATHS and STRING-IMPORT-PATHS to flycheck-dmd variables."
+  (setq flycheck-dmd-include-path import-paths)
+  (let ((flags (mapcar #'(lambda (x) (concat "-J" x)) string-import-paths)))
+    (setq flycheck-dmd-args flags)))
+
+(defun fldd--cache-is-updated-p ()
+  "Return non-nil if `fldd--cache-file' is up-to-date."
+  (let ((conf-timestamp (fldd--get-timestamp "dub.selections.json"))
+        (cache-timestamp (fldd--get-timestamp fldd--cache-file)))
+    (and conf-timestamp cache-timestamp
+         (time-less-p conf-timestamp cache-timestamp))))
+
+(defvar fldd--cache-file ".fldd.cache"
+  "File to cache the result of dub describe.")
+
+;;;###autoload
+(defcustom flycheck-dmd-dub-use-cache-p nil
+  "Non-nil means that `flycheck-dmd-dub-set-variables' reuses the result of dub describe by using cache file.")
```

Comment:
I added `flycheck-dmd-dub` group and add `flycheck-dmd-dub-use-cache-p` to it.
I also added `:type boolean` to the custom variable.